python/pyiceberg/typedef.py

```diff
@@ -126,30 +135,41 @@ def json(self, exclude_none: bool = True, exclude: Optional[Set[str]] = None, by
 
 
 class PydanticStruct(IcebergBaseModel):
```

Comment:
If we can't use Pydantic for defaults, is there still value in using it as the basis for our structs?

Reply:
This is not required anymore, I've refactored this, and moved everything into a single `Record` class, since less is more